internal/graph/check_fast_path.go

```diff
@@ -557,16 +557,11 @@ func (c *LocalChecker) resolveFastPath(ctx context.Context, leftChans []chan *it
 	defer func() {
 		cancel()
 		iter.Stop()
-		if !leftOpen {
-			return
-		}
-		go func() {
-			for msg := range leftChan {
-				if msg.iter != nil {
-					msg.iter.Stop()
-				}
+		for msg := range leftChan {
```

Comment:
Is the bug due to invalid check !leftOpen or the check for leftChan is inside go func() ?

Reply:
Im a little confused. How is this a leak. 

And the whole purpose was to not block yielding the result because this will affect your latency on short circuited resolutions.

Reply:
It's a leak in the tests.

> And the whole purpose was to not block yielding the result because this will affect your latency on short circuited resolutions.

I understand the goal, but then we need to figure out a way of achieving it without a goroutine that nobody is waiting on.

Do we have evidence that latency is majorly affected by this? Also: doesn't the existing context cancellation code prevent the remaining iterators from doing a lot of unnecessary work once the signal has been sent?

Reply:
So a "leak" in the tests, which is not a real leak in actual runtime because it actually gets reaped, just the tooling doing a false positive?

> that latency is majorly affected by this?

If the context cancellation happens right after the request has been fired of to the DS you have to wait for the I/O (depending on the driver, which as you know there are drivers that don't handle well the context cancellations) so that I/O could be from microseconds to 10s of milliseconds depending on the conditions.